eth/sync.go

```diff
@@ -228,10 +228,13 @@ func (pm *ProtocolManager) synchronise(ctx context.Context, peer *peer) {
 	}
 	// Make sure the peer's TD is higher than our own
 	currentBlock := pm.blockchain.CurrentBlock()
-	td := pm.blockchain.GetTd(currentBlock.Hash(), currentBlock.NumberU64())
+	hash := currentBlock.Hash()
+	td := pm.blockchain.GetTd(hash, currentBlock.NumberU64())
 
 	pHead, pTd := peer.Head()
-	if pTd.Cmp(td) <= 0 {
+	if cmp := pTd.Cmp(td); cmp < 0 {
```

Comment:
Perhaps this whole fix can be reduced to just changing the `<=` to a `<`. That way the 'best' choice being arbitrary doesn't matter much because we'll still sync up with anyone at the same TD, and when the actual blocks are pulled then our normal tie-breaker logic will set things straight. This should make things *eventually* resolve, even if it takes a few re-syncs with different peers.

Reply:
#304 expands on this